packages/core/src/animation/Animator.ts

```diff
@@ -59,10 +59,11 @@ export class Animator extends Component {
   }
 
   set animatorController(animatorController: AnimatorController) {
-    if (animatorController !== this._animatorController) {
+    if (animatorController && animatorController !== this._animatorController) {
       this._controllerUpdateFlag && this._controllerUpdateFlag.destroy();
       this._controllerUpdateFlag = animatorController && animatorController._registerChangeFlag();
       this._animatorController = animatorController;
+      this._reset();
```

Comment:
I check `_reset` func, maybe use `this._animationCurveOwners`  to revert is better